kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/StringKekid.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault;
+
+import java.util.Objects;
+
+import io.kroxylicious.kms.service.KekId;
+import io.kroxylicious.kms.service.KmsException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public record StringKekid(@NonNull String keyId) implements KekId {
+
+    public StringKekid {
```

Comment:
@k-wall @robobario would this be better called `VaultKekId`. I know it is a string but its the vault specific implementation of an ID.